### ISSUE TITLE
Add xgboost as engine for linear_reg()

### ISSUE THREAD

**Issue by jaredlander** [tidymodels/parsnip#118]
`{xgboost}` can fit boosted, penalized linear models by setting `booster="gblinear"`. This would be a great addition to `linear_reg()`, or perhaps there can be a `boost_linear_reg()` function and it can also include `gbm()`.

Making this addition can probably take advantage of existing `xgb.train()` implementation because it only requires changing `booster` then using the `lambda` `lambda_bias` and `alpha` parameters consistently with the `penalty` and `mixture` arguments.

**Comment by martinocrippa** [comment]
Hi @juliasilge ,
as I wrote via e-mail, it could be useful to implement an xgboost model for regression with a linear booster. 
My target to predict derive from count or can be observed in a percentage way, I think it's better to use an ensemble model with a linear booster to test different distribution.
My needs are to change the loss function because my target is skewness distributed, to manage it would be amazing if I will use linear booster parameters. 
thanks 
have a nice day
MC

**Comment by jcutler79** [comment]
I would love to see a linear booster option added as well, thus enabling tuning for the three parameters listed with the linear booster in the xgboost package reference manual.
